Drivers/CAN/can.h

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file can.h
+ * @author Hamza Iqbal (iqbal.ha@northeastern.edu)
+ * @brief CAN Driver
+ * @date 2023-06-14
+ * 
+ * https://www.st.com/resource/en/user_manual/um1725-description-of-stm32f4-hal-and-lowlayer-drivers-stmicroelectronics.pdf
+ * CAN Documentation can be found in pages 87-113
+ */
+
+#ifndef CAN_DRIVER_H
+#define CAN_DRIVER_H
+
+#include "stm32f4xx_hal.h"
+#include "can_config.h"
+
+
+
+/*
+    Type Definitions
+    *************************************************************
+*/
+typedef struct
+{
+    CAN_HandleTypeDef* CAN_handler;
+    CAN_TxHeaderTypeDef* CAN_tx_header;
+    CAN_RxHeaderTypeDef* CAN_rx_header;
+    uint32_t id = 0;          // can identifier
+    uint8_t len = 8;      `   // length of data
+    uint8_t data[8] = { 0 };  // data
+    uint8_t bus = 0;          // used to identify where the message came from when events() is used.
+    
+} CAN_t;
```

Comment:
So each CAN message is loaded with a specific CAN bus by default? I kinda like that idea a lil hmmm